src/compiler/abstract_compiler.nit

```diff
@@ -135,7 +135,8 @@ end
 class Toolchain
 	var toolcontext: ToolContext
 
-	fun compile_dir: String
+	# Directory where to generate all C files
+	fun compile_dir(compiler: AbstractCompiler): String
```

Comment:
Je me demande si a ce niveau ca serait pas plus simple d'ajouter un champ `compiler` à la classe plutot qu'un parametre systematique a toutes les methodes. J'aime pas les associations bidirectionnelles one-to-one en général mais ici ca peut avoir du sens pour la gestion de la combinatoires des onfigurations.